### ISSUE TITLE
METADATA_TRAITS settings.py change in 5.12 breaks reading of a written-to-yaml transformed resource descriptor 

### ISSUE THREAD

**Issue by hjoukl** [frictionlessdata/frictionless-py#1514]
# Overview

Up to versions <= 5.11.1:

Describing a resource from a given csv file, transforming it (e.g. adding a field), writing the transformed resource description to a yaml file and then reading this description again worked.

> 5.11.1:
This breaks since 5.12.1 (probably 5.12.0).  Looks like this change in `settings.py` is the culprit: 

Blame:
https://github.com/frictionlessdata/framework/blame/main/frictionless/settings.py#L104

Commit:
https://github.com/frictionlessdata/framework/commit/fe2165b2c63c33a4b6c4e3ee216ba143bc1be691

The reason seems to be that transforming the original resource removes the `path` property from the resource description.

# Example

Data file colors.csv:
```
no,color,description
1,red,The color of blood
2,green,The color of hope
3,blue,The color of oceans
4,yellow,Bright as the sun
5,black,Dark as the night
```

Sample code transformed_resource_yaml_reread.py:

``` 
from pprint import pprint
from frictionless import Resource, describe, transform, steps

source = describe("colors.csv")
print('Original resource:')
pprint(source)

target = transform(
    source,
    steps=[
        steps.field_add(name="note", value="eu", descriptor={"type": "string"}),
    ]
)
print('Transformed resource:')
pprint(target)
target.to_yaml("colors.yaml")

reread_res = Resource("colors.yaml")
print('Reread resource:')
print(reread_res)
```

5.11.1 output:
```
$ ./frictionless-5.11.1-venv/bin/python transformed_resource_yaml_reread.py 
Original resource:
{'name': 'colors',
 'type': 'table',
 'path': 'colors.csv',
 'scheme': 'file',
 'format': 'csv',
 'mediatype': 'text/csv',
 'encoding': 'utf-8',
 'schema': {'fields': [{'name': 'no', 'type': 'integer'},
                       {'name': 'color', 'type': 'string'},
                       {'name': 'description', 'type': 'string'}]}}
Transformed resource:
{'name': 'colors',
 'type': 'table',
 'data': [],
 'scheme': '',
 'format': 'inline',
 'mediatype': 'text/csv',
 'extrapaths': [],
 'schema': {'fields': [{'name': 'no', 'type': 'integer'},
                       {'name': 'color', 'type': 'string'},
                       {'name': 'description', 'type': 'string'},
                       {'name': 'note', 'type': 'string'}]}}
Reread resource:
{'name': 'colors',
 'type': 'table',
 'data': [],
 'format': 'inline',
 'mediatype': 'text/csv',
 'extrapaths': [],
 'schema': {'fields': [{'name': 'no', 'type': 'integer'},
                       {'name': 'color', 'type': 'string'},
                       {'name': 'description', 'type': 'string'},
                       {'name': 'note', 'type': 'string'}]}}
```

5.12.1 output:
```
$ ./frictionless-5.12.1-venv/bin/python transformed_resource_yaml_reread.py 
Original resource:
{'name': 'colors',
 'type': 'table',
 'path': 'colors.csv',
 'scheme': 'file',
 'format': 'csv',
 'mediatype': 'text/csv',
 'encoding': 'utf-8',
 'schema': {'fields': [{'name': 'no', 'type': 'integer'},
                       {'name': 'color', 'type': 'string'},
                       {'name': 'description', 'type': 'string'}]}}
Transformed resource:
{'name': 'colors',
 'type': 'table',
 'data': [],
 'scheme': '',
 'format': 'inline',
 'mediatype': 'text/csv',
 'extrapaths': [],
 'schema': {'fields': [{'name': 'no', 'type': 'integer'},
                       {'name': 'color', 'type': 'string'},
                       {'name': 'description', 'type': 'string'},
                       {'name': 'note', 'type': 'string'}]}}
Reread resource:
{'name': 'colors',
 'type': 'json',
 'path': 'colors.yaml',
 'scheme': 'file',
 'format': 'yaml',
 'mediatype': 'text/yaml'}
```

As you can see the 5.12.1 version completely misinterprets the `colors.yaml` resource description. This happens due to the `settings.py` change that removed the `data` property from the `METADATA_TRAITS` resource properties.

This happens silently i.e. writing the resource description and re-reading it succeeds but after reading the resource
description is completely changed. Which you only realize when trying to make use of the missing/altered information.

It is not mentioned as a breaking change in the changelog.

Unfortunately, I don't have a good grasp on what *should* happen here. 

Best regards,
Holger

**Comment by roll** [comment]
Thanks! We will fix it

**Comment by roll** [comment]
Thanks @hjoukl! It's fixed in https://github.com/frictionlessdata/framework/pull/1539

In-general, there is a bigger problem with the legacy API we preserve for backward compatibility. In a few words, it's never 100% safe to use `Resource('descriptor.yaml')` as it involves a guessing step, while a safe way is to use `Resource.from_descriptor('descriptor.yaml')`. Similarly, instead of `Resource('table.json')` one has to use `Resource(path='table.json')`.

We're thinking about how to improve this situation

**Comment by hjoukl** [comment]
> Thanks @hjoukl! It's fixed in #1539

Great, thanks! Will give it a spin.

> In-general, there is a bigger problem with the legacy API we preserve for backward compatibility. In a few words, it's never 100% safe to use `Resource('descriptor.yaml')` as it involves a guessing step, while a safe way is to use `Resource.from_descriptor('descriptor.yaml')`. Similarly, instead of `Resource('table.json')` one has to use `Resource(path='table.json')`.
> 
> We're thinking about how to improve this situation

Ah, I didn't realize I was using legacy API and must have overlooked the `Resource.from_descriptor` factory. For my use case I could simply switch to using this instead.

Best regards,
Holger
